### PR TITLE
React Rendering Techniques: Prevent markdown issue/inconsistency

### PR DESCRIPTION
**Pull request by fadingmorsecode**
## Because

Prevents markdown from highlighting the word "in" and keeps the wording consistent with another div element that already uses the word "on". 

## This PR

* Replaces two instances of "in" with "on"

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Thanks for opening this PR @fadingmorsecode
We can't change from `in` to `on` because that would be incorrect grammar.
Instead, the actual fix would be changing the codeblock language from `javascript` to `jsx` (it doesn't recognise the JSX syntax using `javascript` highlighting, so takes the `in` word as the `in` operator).

Would you be able to do the following the fix this lesson up:
1. Revert `on` back to `in`.
2. Change **all** codeblocks' language from `javascript` to `jsx` **except** the code block from lines 120-126, as that one actually does need to be `javascript`.
3. Remove the rogue blank lines on line 102 and line 125.
4. Follow the curriculum repo [linting instructions](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md) and run `npm run fix:lesson -- react/getting_started_with_react/rendering_techniques.md` after everything else?

Doing those 4 steps should fix all the other linting errors as well. Let me know if you run into any issues with this.

**Comment by fadingmorsecode** [comment]
> Thanks for opening this PR @fadingmorsecode We can't change from `in` to `on` because that would be incorrect grammar. Instead, the actual fix would be changing the codeblock language from `javascript` to `jsx` (it doesn't recognise the JSX syntax using `javascript` highlighting, so takes the `in` word as the `in` operator).
> 
> Would you be able to do the following the fix this lesson up:
> 
> 1. Revert `on` back to `in`.
> 2. Change **all** codeblocks' language from `javascript` to `jsx` **except** the code block from lines 120-126, as that one actually does need to be `javascript`.
> 3. Remove the rogue blank lines on line 102 and line 125.
> 4. Follow the curriculum repo [linting instructions](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md) and run `npm run fix:lesson -- react/getting_started_with_react/rendering_techniques.md` after everything else?
> 
> Doing those 4 steps should fix all the other linting errors as well. Let me know if you run into any issues with this.

Before I go ahead and make those changes, I would like to point out that on line 220 I did not make any changes yet the word "on" was used (which is part of why I thought to change the others). So my question is, do I make that an "in" in addition to the requested changes?

**Comment by MaoShizhong** [comment]
@fadingmorsecode no worries - the lines that say "on the list" should all be changed to "in the list". The other changes requested will resolve the other things that will also result in the correct syntax highlighting.

File: react/getting_started_with_react/rendering_techniques.md [hunk 1]
```diff
@@ -297,7 +298,7 @@ function List(props) {
           })}
         </ul>
       )}
-      {props.animals && props.animals.length === 0 && <div>There are no animals in the list!</div>}
+      {props.animals && props.animals.length === 0 && <div>There are no animals on the list!</div>}
```
Comment (MaoShizhong) [inline react/getting_started_with_react/rendering_techniques.md:301]: Just need to fix this one then we should be good to merge.
```suggestion
      {props.animals && props.animals.length === 0 && <div>There are no animals in the list!</div>}
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]

**Review by MaoShizhong — APPROVED** [review]
Many thanks for contributing!

**Pull request merged**